manticore/core/smtlib/constraints.py

```diff
@@ -163,23 +163,26 @@ def __str__(self):
             elif isinstance(exp, Bool):
                 result += '(declare-fun %s () Bool)' % name
             elif isinstance(exp, Array):
-                result += '(declare-fun %s () (Array (_ BitVec %d) (_ BitVec 8)))' % (name, exp.index_bits)
+                result += '(declare-fun %s () (Array (_ BitVec %d) (_ BitVec %d)))' % (name, exp.index_bits, exp.value_bits)
             else:
                 raise Exception("Type not supported %r", exp)
             result += '(assert (= %s %s))\n' % (name, smtlib)
 
-        r = translator.pop()
-        while r is not None:
-            result += '(assert %s)\n' % r
-            r = translator.pop()
+        constraint_str = translator.pop()
+        while constraint_str is not None:
+            if not constraint_str is 'true':
```

Comment:
maybe `if constraint_str != 'true':`

Reply:
For sure this shouldn't use `is`, see https://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce